### PR TITLE
Product subscriptions: Analytics

### LINKED ISSUE #10163: Add tracking
Add tracking following plan from project thread: pe5sF9-2cm-p2

**Reminder:** Double check that the tracking events are added for simple product and variation product. For example: 
- Add `_product_subscription_view_expiry_date_tapped` in https://github.com/woocommerce/woocommerce-android/blob/a8ded567c9e35a260baa7edc2d89ecb7b1683bb0/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailCardBuilder.kt#L848

- And add again `_product_subscription_view_expiry_date_tapped` in https://github.com/woocommerce/woocommerce-android/blob/a8ded567c9e35a260baa7edc2d89ecb7b1683bb0/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationDetailCardBuilder.kt#L376

For references but better to check the P2 tracking plan to see the latest: 

Screenshot 2023-11-16 at 17 13 48

### PR DESCRIPTION
**Pull request by 0nko**
Resolves #10163. The PR adds analytics for the following events:

```
PRODUCT_DETAILS_VIEW_SUBSCRIPTION_EXPIRATION_TAPPED
PRODUCT_DETAILS_VIEW_SUBSCRIPTION_FREE_TRIAL_TAPPED
PRODUCT_VARIATION_VIEW_SUBSCRIPTION_EXPIRATION_TAPPED
PRODUCT_VARIATION_VIEW_SUBSCRIPTION_FREE_TRIAL_TAPPED
PRODUCT_SUBSCRIPTION_EXPIRATION_DONE_BUTTON_TAPPED
PRODUCT_SUBSCRIPTION_FREE_TRIAL_DONE_BUTTON_TAPPED
```

Updated events:
```
PRODUCT_LIST_LOADED(is_eligible_for_subscriptions)
ADD_PRODUCT_PRODUCT_TYPE_SELECTED(product_type) // new types added automatically
```

Note: Please merge https://github.com/woocommerce/woocommerce-android/pull/10176 first.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt [hunk 1]
```diff
@@ -585,7 +587,11 @@ enum class AnalyticsEvent(val siteless: Boolean = false) {
 
     // -- Product subscriptions
     PRODUCT_DETAILS_VIEW_SUBSCRIPTIONS_TAPPED,
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt]: I think this event should be removed, as we are getting rid of the `subscriptions` read only section. Anyway, we can probably handle this in the task where we remove all of that.

**Comment by JorgeMucientes** [comment]
Hey @0nko changes look good. However there's some discussion ongoing related to tracking you might want to check before merging: p1700132234806529-slack-C03L1NF1EA3

Let us know your thoughts.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt]: Yeah, that was my thought.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Thank you for applying the changes @0nko 👍🏼 
Ready to merge :shipit:

**Pull request merged**